MatrixKit/Models/Room/MXKRoomDataSource.m

```diff
@@ -665,7 +668,18 @@ - (void)listenTypingNotifications
         if (direction == MXEventDirectionForwards)
         {
             // Retrieve typing users list
-            NSMutableArray *typingUsers = [NSMutableArray arrayWithArray:_room.typingUsers];
+            NSMutableArray *typingUsers;
+            
+            if (self.showTypingNotifications)
```

Comment:
I suggest adding a setter for 'showTypingNotifications' in order to remove typingNotifListener when this flag is NO.
BTW, the whole code should be updated to support currentTypingUsers = nil.